### ISSUE TITLE
Errors with TypeScript and --isolatedModules compile option

### ISSUE THREAD

**Issue by petrovi4** [sequelize/sequelize#10575]
I use Babel to compile to native js, and TypeScript for type-checking.

## What are you doing?

- tsconfig.json
```js
{
	"compilerOptions": {
...
		// Disallow features that require cross-file information for emit.
		// Cross-file information is not supported by Babel.
		"isolatedModules": true,
...
	},
	"include": ["."],
	"exclude": ["node_modules"]
}
```

## What is actually happening?
tsc Compilation errors

```
node_modules/sequelize/types/lib/transaction.d.ts:62:14 - error TS1209: Ambient const enums are not allowed when the '--isolatedModules' flag is provided.

62   const enum ISOLATION_LEVELS {
                ~~~~~~~~~~~~~~~~

node_modules/sequelize/types/lib/transaction.d.ts:69:14 - error TS1209: Ambient const enums are not allowed when the '--isolatedModules' flag is provided.

69   const enum TYPES {
                ~~~~~

node_modules/sequelize/types/lib/transaction.d.ts:110:14 - error TS1209: Ambient const enums are not allowed when the '--isolatedModules' flag is provided.

110   const enum LOCK {
                 ~~~~

[6:32:25 PM] Found 3 errors. Watching for file changes.
```

Tested with latest release: "sequelize": "^5.1.0"

**Comment by SimonSchick** [comment]
Ah yea, we didn't test that with Babel, I think it's reasonable to change these to be non const.

**Comment by SimonSchick** [comment]
Can you locally edit the definition files and see if this change fixes it?
#10585

**Comment by SimonSchick** [comment]
Fixed via #10585
